From 20d623401f47b037ba9fd5038e2bb5ae15274791 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Mon, 19 Oct 2026 05:54:44 -0700
Subject: [PATCH] test: move the wallet tests' extra runs to variants

tool_wallet.py and the wallet_* tests now declare their descriptors and
cli runs as variants instead of TEST_PARAMS entries in test_runner.py.
Their results and timing.json entries are named "<file>[descriptors]"
and "<file>[cli]" instead of "<file> --descriptors" and
"<file> --usecli".

The full functional suite runs the same tests with the same results.
---
 test/functional/test_runner.py              | 15 --------
 test/functional/timing.json                 | 42 ++++++++++-----------
 test/functional/tool_wallet.py              |  2 +
 test/functional/wallet_address_types.py     |  2 +
 test/functional/wallet_assumeutxo.py        |  2 +
 test/functional/wallet_avoidreuse.py        |  2 +
 test/functional/wallet_balance.py           |  2 +
 test/functional/wallet_createwallet.py      |  6 +++
 test/functional/wallet_encryption.py        |  2 +
 test/functional/wallet_hd.py                |  2 +
 test/functional/wallet_importprunedfunds.py |  2 +
 test/functional/wallet_keypool_topup.py     |  2 +
 test/functional/wallet_labels.py            |  2 +
 test/functional/wallet_listsinceblock.py    |  2 +
 test/functional/wallet_listtransactions.py  |  2 +
 test/functional/wallet_multiwallet.py       |  6 +++
 test/functional/wallet_watchonly.py         |  2 +
 17 files changed, 59 insertions(+), 36 deletions(-)

diff --git a/test/functional/test_runner.py b/test/functional/test_runner.py
index 053e72ee7fa506137ce00af06939ce8da3716ebd..55ce03d05cbc8d42c62f1fbd26dfc4fdf2891daa 100755
--- a/test/functional/test_runner.py
+++ b/test/functional/test_runner.py
@@ -132,25 +132,10 @@ TEST_PARAMS = {
     #    testName --param1 --param2
     #    testname --param3
     # FIXME: "rpc_psbt.py": [["--descriptors"]],
-    "wallet_address_types.py": [["--descriptors"]],
-    "tool_wallet.py": [["--descriptors"]],
-    "wallet_avoidreuse.py": [["--descriptors"]],
-    "wallet_balance.py": [["--descriptors"]],
     # FIXME: "wallet_basic.py": [["--descriptors"]],
-    "wallet_createwallet.py": [["--usecli"], ["--descriptors"]],
-    "wallet_encryption.py": [["--descriptors"]],
-    "wallet_hd.py": [["--descriptors"]],
-    "wallet_assumeutxo.py": [["--descriptors"]],
-    "wallet_importprunedfunds.py": [["--descriptors"]],
     # FIXME: "wallet_keypool.py": [["--descriptors"]],
-    "wallet_keypool_topup.py": [["--descriptors"]],
-    "wallet_labels.py": [["--descriptors"]],
-    "wallet_listsinceblock.py": [["--descriptors"]],
-    "wallet_listtransactions.py": [["--descriptors"]],
-    "wallet_multiwallet.py": [["--usecli"], ["--descriptors"]],
     "wallet_txn_doublespend.py": [["--mineblock"]],
     "wallet_txn_clone.py": [["--mineblock"]],
-    "wallet_watchonly.py": [["--usecli"]],
 }
 
 # Used to limit the number of tests, when list of tests is not provided on command line
diff --git a/test/functional/timing.json b/test/functional/timing.json
index fac904befdbd37e019d18f540741f804688bbe5e..623b5e3a90e86c6b8cd28a19433c66099678a798 100644
--- a/test/functional/timing.json
+++ b/test/functional/timing.json
@@ -1028,7 +1028,7 @@
   "time": 7
  },
  {
-  "name": "tool_wallet.py --descriptors",
+  "name": "tool_wallet.py[descriptors]",
   "time": 4
  },
  {
@@ -1040,7 +1040,7 @@
   "time": 24
  },
  {
-  "name": "wallet_address_types.py --descriptors",
+  "name": "wallet_address_types.py[descriptors]",
   "time": 24
  },
  {
@@ -1048,7 +1048,7 @@
   "time": 9
  },
  {
-  "name": "wallet_assumeutxo.py --descriptors",
+  "name": "wallet_assumeutxo.py[descriptors]",
   "time": 16
  },
  {
@@ -1056,7 +1056,7 @@
   "time": 17
  },
  {
-  "name": "wallet_avoidreuse.py --descriptors",
+  "name": "wallet_avoidreuse.py[descriptors]",
   "time": 33
  },
  {
@@ -1068,7 +1068,7 @@
   "time": 9
  },
  {
-  "name": "wallet_balance.py --descriptors",
+  "name": "wallet_balance.py[descriptors]",
   "time": 9
  },
  {
@@ -1088,12 +1088,12 @@
   "time": 9
  },
  {
-  "name": "wallet_createwallet.py --descriptors",
-  "time": 11
+  "name": "wallet_createwallet.py[cli]",
+  "time": 9
  },
  {
-  "name": "wallet_createwallet.py --usecli",
-  "time": 9
+  "name": "wallet_createwallet.py[descriptors]",
+  "time": 11
  },
  {
   "name": "wallet_descriptor.py",
@@ -1112,7 +1112,7 @@
   "time": 6
  },
  {
-  "name": "wallet_encryption.py --descriptors",
+  "name": "wallet_encryption.py[descriptors]",
   "time": 7
  },
  {
@@ -1124,7 +1124,7 @@
   "time": 17
  },
  {
-  "name": "wallet_hd.py --descriptors",
+  "name": "wallet_hd.py[descriptors]",
   "time": 18
  },
  {
@@ -1148,7 +1148,7 @@
   "time": 4
  },
  {
-  "name": "wallet_importprunedfunds.py --descriptors",
+  "name": "wallet_importprunedfunds.py[descriptors]",
   "time": 11
  },
  {
@@ -1160,7 +1160,7 @@
   "time": 16
  },
  {
-  "name": "wallet_keypool_topup.py --descriptors",
+  "name": "wallet_keypool_topup.py[descriptors]",
   "time": 23
  },
  {
@@ -1168,7 +1168,7 @@
   "time": 10
  },
  {
-  "name": "wallet_labels.py --descriptors",
+  "name": "wallet_labels.py[descriptors]",
   "time": 11
  },
  {
@@ -1180,7 +1180,7 @@
   "time": 9
  },
  {
-  "name": "wallet_listsinceblock.py --descriptors",
+  "name": "wallet_listsinceblock.py[descriptors]",
   "time": 18
  },
  {
@@ -1188,7 +1188,7 @@
   "time": 7
  },
  {
-  "name": "wallet_listtransactions.py --descriptors",
+  "name": "wallet_listtransactions.py[descriptors]",
   "time": 9
  },
  {
@@ -1196,12 +1196,12 @@
   "time": 32
  },
  {
-  "name": "wallet_multiwallet.py --descriptors",
-  "time": 34
+  "name": "wallet_multiwallet.py[cli]",
+  "time": 33
  },
  {
-  "name": "wallet_multiwallet.py --usecli",
-  "time": 33
+  "name": "wallet_multiwallet.py[descriptors]",
+  "time": 34
  },
  {
   "name": "wallet_pruning.py",
@@ -1248,7 +1248,7 @@
   "time": 3
  },
  {
-  "name": "wallet_watchonly.py --usecli",
+  "name": "wallet_watchonly.py[cli]",
   "time": 3
  }
 ]
diff --git a/test/functional/tool_wallet.py b/test/functional/tool_wallet.py
index 4009feee045d4001fa5b57d5fc7d4ea3632ebbb3..694cdc854ebadff5faa21557980f151bffc00cbe 100644
--- a/test/functional/tool_wallet.py
+++ b/test/functional/tool_wallet.py
@@ -16,6 +16,8 @@ BUFFER_SIZE = 16 * 1024
 
 
 class ToolWalletTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
diff --git a/test/functional/wallet_address_types.py b/test/functional/wallet_address_types.py
index 53549d7854ff4bf5381de4d06445f77eff1164ee..5bc98a9e6a971237a7083ccaccbd77817aeea9fc 100644
--- a/test/functional/wallet_address_types.py
+++ b/test/functional/wallet_address_types.py
@@ -41,6 +41,8 @@ from test_framework.util import assert_equal, assert_greater_than
 
 
 class AddressTypeTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.num_nodes = 5
         self.noban_tx_relay = True
diff --git a/test/functional/wallet_assumeutxo.py b/test/functional/wallet_assumeutxo.py
index 3e235093f50b901b587efb5d3dacb2551eeabdf8..cf85b647efd74a471c903fcdc8dde2cc973ee205 100644
--- a/test/functional/wallet_assumeutxo.py
+++ b/test/functional/wallet_assumeutxo.py
@@ -23,6 +23,8 @@ FINAL_HEIGHT = 399
 
 
 class AssumeutxoTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
diff --git a/test/functional/wallet_avoidreuse.py b/test/functional/wallet_avoidreuse.py
index a9540da375a9a9e95a3a155d166a6b002eea0f30..48556f90719e00432f4fe4918de0bd107ef5fc7f 100644
--- a/test/functional/wallet_avoidreuse.py
+++ b/test/functional/wallet_avoidreuse.py
@@ -75,6 +75,8 @@ def assert_balances(node, mine):
 
 
 class AvoidReuseTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.num_nodes = 2
         self.noban_tx_relay = True
diff --git a/test/functional/wallet_balance.py b/test/functional/wallet_balance.py
index 19d02f0ec4ac5505e77799b638ba909aa84abc50..a0db305b06d8d577e0077c33d112697ff7d9dad9 100644
--- a/test/functional/wallet_balance.py
+++ b/test/functional/wallet_balance.py
@@ -43,6 +43,8 @@ def create_transactions(node, address, amt, fees):
 
 
 class WalletTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.num_nodes = 2
         self.setup_clean_chain = True
diff --git a/test/functional/wallet_createwallet.py b/test/functional/wallet_createwallet.py
index 501e4202b8875f7b557d57fd5cc1e8199d3635f9..cdc73926e8a84401fc6dfce9f4a05f6224c423e6 100644
--- a/test/functional/wallet_createwallet.py
+++ b/test/functional/wallet_createwallet.py
@@ -12,6 +12,12 @@ from test_framework.wallet_util import bytes_to_wif, generate_wif_key
 
 
 class CreateWalletTest(BitcoinTestFramework):
+    variants = {
+        "default": {},
+        "cli": {"usecli": True},
+        "descriptors": {"descriptors": True},
+    }
+
     def set_test_params(self):
         self.num_nodes = 1
 
diff --git a/test/functional/wallet_encryption.py b/test/functional/wallet_encryption.py
index e927dfa54ff92de13359b6f0b9728991f398b8f8..7a6823ac5fa68debea7f57aa3d7fd6be1a28c7ec 100644
--- a/test/functional/wallet_encryption.py
+++ b/test/functional/wallet_encryption.py
@@ -15,6 +15,8 @@ from test_framework.util import (
 
 
 class WalletEncryptionTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
diff --git a/test/functional/wallet_hd.py b/test/functional/wallet_hd.py
index 7ad110e9950d6a2a37e18e587e50351043aaa9cf..19295920f8e0b39dea0117b9f4a30561ee8afd42 100644
--- a/test/functional/wallet_hd.py
+++ b/test/functional/wallet_hd.py
@@ -12,6 +12,8 @@ from test_framework.util import assert_equal, assert_raises_rpc_error
 
 
 class WalletHDTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 2
diff --git a/test/functional/wallet_importprunedfunds.py b/test/functional/wallet_importprunedfunds.py
index 783fce9df7b77fee7d09ff59303557e3b7bd3230..280ca2ed4fb414a7f32dfd2f3debf8af1d5a250a 100644
--- a/test/functional/wallet_importprunedfunds.py
+++ b/test/functional/wallet_importprunedfunds.py
@@ -14,6 +14,8 @@ from test_framework.wallet_util import bytes_to_wif
 
 
 class ImportPrunedFundsTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 2
diff --git a/test/functional/wallet_keypool_topup.py b/test/functional/wallet_keypool_topup.py
index d8ac98710adf9325725fe4a7090d22191cf5877e..9aa0716c43773d6404c1517bb29243ed87777e41 100644
--- a/test/functional/wallet_keypool_topup.py
+++ b/test/functional/wallet_keypool_topup.py
@@ -19,6 +19,8 @@ from test_framework.util import assert_equal
 
 
 class KeypoolRestoreTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 2
diff --git a/test/functional/wallet_labels.py b/test/functional/wallet_labels.py
index 4192a5822c0a1d68c08e4199b37221e90f617803..86da66f4195c9f9e98ac7395af5d6e7835cd5b14 100644
--- a/test/functional/wallet_labels.py
+++ b/test/functional/wallet_labels.py
@@ -18,6 +18,8 @@ from test_framework.wallet_util import test_address
 
 
 class WalletLabelsTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
diff --git a/test/functional/wallet_listsinceblock.py b/test/functional/wallet_listsinceblock.py
index 8bab3ce7be467e85225ec6956a577e106be9e0ba..1bc03388a181285211e276e2e17619e38a150c4a 100644
--- a/test/functional/wallet_listsinceblock.py
+++ b/test/functional/wallet_listsinceblock.py
@@ -17,6 +17,8 @@ from test_framework.wallet_util import bytes_to_wif
 
 
 class ListSinceBlockTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.num_nodes = 4
         self.setup_clean_chain = True
diff --git a/test/functional/wallet_listtransactions.py b/test/functional/wallet_listtransactions.py
index 388e6a7127008339ea62879e975487fe857a9ef9..beb32daead7fa3429c5a150df621c7901977bd64 100644
--- a/test/functional/wallet_listtransactions.py
+++ b/test/functional/wallet_listtransactions.py
@@ -10,6 +10,8 @@ from test_framework.util import assert_array_result, assert_equal
 
 
 class ListTransactionsTest(BitcoinTestFramework):
+    variants = {"default": {}, "descriptors": {"descriptors": True}}
+
     def set_test_params(self):
         self.num_nodes = 2
         self.noban_tx_relay = True
diff --git a/test/functional/wallet_multiwallet.py b/test/functional/wallet_multiwallet.py
index 3732a15415dd4e41f385548d5afb5f0b433c9fd9..5b442430c34cdf87ccc04e837cbc7b3e0d7ecde1 100644
--- a/test/functional/wallet_multiwallet.py
+++ b/test/functional/wallet_multiwallet.py
@@ -48,6 +48,12 @@ def test_load_unload(node, name, timeout=60.0):
 
 
 class MultiWalletTest(BitcoinTestFramework):
+    variants = {
+        "default": {},
+        "cli": {"usecli": True},
+        "descriptors": {"descriptors": True},
+    }
+
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 2
diff --git a/test/functional/wallet_watchonly.py b/test/functional/wallet_watchonly.py
index 2e73e4841542890dead3fb15dc83527992c8f1b0..a398d4505e09a2cf80eba983399fa9c3104dc807 100644
--- a/test/functional/wallet_watchonly.py
+++ b/test/functional/wallet_watchonly.py
@@ -9,6 +9,8 @@ from test_framework.util import assert_equal, assert_raises_rpc_error
 
 
 class CreateWalletWatchonlyTest(BitcoinTestFramework):
+    variants = {"default": {}, "cli": {"usecli": True}}
+
     def set_test_params(self):
         self.num_nodes = 1
 
-- 
2.50.1 (Apple Git-155)

